### ISSUE TITLE
submitting a new issue , because same issue was from 2017

### ISSUE THREAD

**Issue by gorxk** [polychromatic/polychromatic#228]
submitting a new issue , because same issue was from 2017

I have a razerkeyboard between the demon and this the keys light up. But making macros sucks ass. On a lot of someones plugs and screen shots there's a menu with actual macros and what ever listed. 
As it is now I have no way of knowing if it's working. It doesn't show any kind macro menu, it doesn't show what buttons it saw and to put that macro onto a button, and test it to see if it works.

Expected to workness:
Macro Menu shows buttons pressed, and what button to put that macro on.

Actual Workingness: No menu to make macros -_-

**Comment by lah7** [comment]
I understand. Neither OpenRazer or Polychromatic properly support macros because the projects are more focused on controlling lighting for now. Other projects exist for that functionality, as it's not a Razer-specific thing. 

OpenRazer daemon does supports on-the-fly macros if your keyboard has M1-M5 keys, which works well "in the moment" when doing work or gaming, but there is no frontend that directly saves/loads these.

The current state (and some suggested applications) can be found in Polychromatic's FAQs. :mag: 

* https://polychromatic.app/docs/faqs/

Macros to me is essentially **key re-mapping**. In order for Polychromatic to support this I need a library that can be easily configured and works to a good standard. I'm not at a point I'm ready to build a pretty interface. It is still the aim, even if it's been 2+ years. 

* OpenRazer have an open pull request: https://github.com/openrazer/openrazer/pull/880
* Polychromatic might implement a frontend to  **[boppreh/keyboard](https://github.com/boppreh/keyboard)** - (#223)

**TL;DR:** Macro support isn't the top priority, but it'll come eventually. Bear in mind this is developed in my free time, so progress can be slow at times.
